ethdb/database.go

```diff
@@ -90,6 +90,14 @@ type AncientReader interface {
 	AncientSize(kind string) (uint64, error)
 }
 
+// AncientBatchReader is the interface for 'batched' or 'atomic' reading.
+type AncientBatchReader interface {
+	AncientReader
+	// ReadAncients runs the given read operation while ensuring that no writes take place
+	// on the underlying freezer.
+	ReadAncients(fn func(AncientReader) error) (err error)
```

Comment:
Perhaps `ReadAncientsLocked`?

Reply:
The `ReadAncients` operation is analoguous to `ModifyAncients` which is the locked/atomic/batched write operation.